oracle/oracle.go

```diff
@@ -444,3 +458,125 @@ func (o *oracleOps) Delete(volumeID string) error {
+		// wait infinitely for operation to complete
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+		defer cancel()
+	}
+
+	instanceGroupSize := int(count)
+
+	//get containerengine client
+	client, err := containerengine.NewContainerEngineClientWithConfigurationProvider(common.DefaultConfigProvider())
+	if err != nil {
+		return err
+	}
+
+	//get nodepool by ID to be updated
+	nodePoolReq := containerengine.ListNodePoolsRequest{CompartmentId: &o.compartmentID, Name: &instanceGroupID, ClusterId: &o.clusterID}
+	nodePools, err := client.ListNodePools(context.Background(), nodePoolReq)
+	if err != nil {
+		return err
+	}
+
+	if len(nodePools.Items) == 0 {
+		return errors.New("No node pool found with name " + instanceGroupID)
+	}
+	numberOfDomains := len(nodePools.Items[0].NodeConfigDetails.PlacementConfigs)
+	totalClusterSize := numberOfDomains * instanceGroupSize
+	logrus.Println("Setting instanceGroupSize to ", totalClusterSize, " in total ", numberOfDomains, " regions.")
+
+	//get all availabliity domain
+	nodePoolPlacementConfigDetails := make([]containerengine.NodePoolPlacementConfigDetails, numberOfDomains)
+
+	for i, placementConfigs := range nodePools.Items[0].NodeConfigDetails.PlacementConfigs {
+		nodePoolPlacementConfigDetails[i].AvailabilityDomain = placementConfigs.AvailabilityDomain
+		nodePoolPlacementConfigDetails[i].SubnetId = placementConfigs.SubnetId
+	}
+
+	//update node pools
+	req := containerengine.UpdateNodePoolRequest{
+		NodePoolId: nodePools.Items[0].Id, //get node pool id
+		UpdateNodePoolDetails: containerengine.UpdateNodePoolDetails{
+			NodeConfigDetails: &containerengine.UpdateNodePoolNodeConfigDetails{
+				Size:             &totalClusterSize,
+				PlacementConfigs: nodePoolPlacementConfigDetails,
+			},
+		},
+	}
+
+	resp, err := client.UpdateNodePool(context.Background(), req)
+	if err != nil {
+		return err
+	}
+
+	err = o.waitNodeStatus(resp.OpcRequestId, resp.OpcWorkRequestId, client)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *oracleOps) waitNodeStatus(opcRequestID, opcWorkRequestID *string, client containerengine.ContainerEngineClient) error {
+	workReq := containerengine.GetWorkRequestRequest{OpcRequestId: opcRequestID,
+		WorkRequestId: opcWorkRequestID}
+
+	f := func() (interface{}, bool, error) {
+		workResp, err := client.GetWorkRequest(context.Background(), workReq)
+		if err != nil {
+			return nil, true, err
+		}
+
+		if workResp.Status == containerengine.WorkRequestStatusSucceeded {
+			return workResp.Status, false, nil
+		}
+
+		logrus.Debugf("Node status is in [%s] state", workResp.Status)
+		return nil, true, fmt.Errorf("Node status is still in [%s] state", workResp.Status)
+	}
+	_, err := task.DoRetryWithTimeout(f, 5*time.Minute, 10*time.Second)
+	return err
+}
+
+func (o *oracleOps) GetInstanceGroupSize(instanceGroupID string) (int64, error) {
+	client, err := containerengine.NewContainerEngineClientWithConfigurationProvider(common.DefaultConfigProvider())
```

Comment:
use `oracleOps. containerEngine` instead of re-initialzing here.

Reply:
done